docs: add TOC for the developer handbook (#80)
eclipse-edc/docs: docs/developer/developer-handbook.md

> # EDC Developer Handbook

Reviewer: we don't need to name "EDC" everywhere
```suggestion
# Developer's Handbook
```

<!-- TOC -->

> * [EDC Developer Handbook](#edc-developer-handbook)

Reviewer: We don't need that auto-generated root
```suggestion
```

  * [Introduction](#introduction)

>   * [Building your first connector](#building-your-first-connector)

Reviewer 2: ```suggestion
  * [Building a distribution](#building-a-distribution)
```

    * [Seed some data](#seed-some-data)
    * [Perform a data transfer](#perform-a-data-transfer)

>   * [The control plane - core concepts](#the-control-plane---core-concepts)

Reviewer 3: I would split core concepts from the control plane

Author: but it specifically talks about core concepts _of_ the control plane

Reviewer 3: some those core concepts are in fact valid for all the components, not only the connector, like state machines, extension model and so on

Author: i don't see much use in explaining generic state machines, we can assume people know what that is. 
I would rather outline the specific ones we have, i.e. contract negotiation and transfer process state machine, which are of the control plane. 
We could list some general principles, like asynchronicity though.

    * [API objects in detail](#api-objects-in-detail)

>     * [State machines of EDC](#state-machines-of-edc)

Reviewer 2: ```suggestion
    * [EDC State machines](#edc-state-machines)
```

    * [The extension model](#the-extension-model)

>     * [Dependency Injection à la EDC](#dependency-injection-à-la-edc)

Reviewer 2: ```suggestion
    * [EDC Dependency Injection](#EDC-dependency-injection)
```

    * [Policy scopes and evaluation](#policy-scopes-and-evaluation)
  * [The data plane](#the-data-plane)
    * [Data plane selectors](#data-plane-selectors)

>     * [Write your own DataSink and DataSource](#write-your-own-datasink-and-datasource)

Reviewer 2: ```suggestion
    * [Writing DataSink and DataSource extensions](#write-datasink-and-datasource-extensions)
```

>     * [Write your own DataSink and DataSource](#write-your-own-datasink-and-datasource)

Reviewer 2: ```suggestion
    * [Writing a DataSink and DataSource extension](#write-a-datasink-and-datasource-extension)
```

    * [The Control API](#the-control-api)

>   * [Advanced concepts](#advanced-concepts)

Reviewer 3: we have chapters for the connector concepts but not for IH, RS and FCC

Author: that's on purpose - according to the outline defined [here](https://github.com/orgs/eclipse-edc/discussions/61#discussioncomment-6656353) each component should discuss their internals in their own README, to which we will link in the two-line overview.

Reviewer 3: ok, but here, in the general docs we're talking about the connector functionality (control plane + data plane = connector). I agree that detailed stuff should stay in the related repository, but an "8 miles high" view of the whole set of components should stay here, like, what's an identity hub and how it interacts with a connector for example

Author: yep - i thought that was the two-line thing

    * [Events and callbacks](#events-and-callbacks)
    * [The EDC JUnit framework](#the-edc-junit-framework)
    * [Automatic documentation](#automatic-documentation)

>     * [Customize the build](#customize-the-build)

Reviewer 2: ```suggestion
    * [Customizing the build](#customizing-the-build)
```

<!-- TOC -->

## Introduction

--> states the intended audience, required pre-existing knowledge, required tools, etc.

> ## Building your first connector

Reviewer 2: ```suggestion
## Building a distribution```

--> link to one of the samples

> 

Reviewer 4: Personally, I would put this part at the end of the document, but for an overview it may make sense at this point as well.

Author: my thinking was, that it's a developer's handbook, and developers usually want to play around with something, even before reading a bunch of text.

### Seed some data

--> use postman, or even just curl

### Perform a data transfer
--> actually launches another connector, shows a data exchange -> link to samples

## The control plane - core concepts

### API objects in detail

--> explains Assets, Policies, Contract Definitions, etc. from an external API perspective. mentions JSON-LD and related
specs (ODRL, DCAT)

> ### State machines of EDC

Reviewer 2: ```suggestion
### EDC state machines
```

--> gives an overview of the state machines to understand asynchronicity and why some APIs only return IDs

> ### The extension model

Reviewer 4: Would be great to have an overview of all extension points (at least groups)

Author: there is a lot of them, and we can already convert the autodoc manifest to markdown, so I would link that MD file

Reviewer 4: There are many indeed, and so an overview might be useful for developers. The suggested link to the manifest should work well. Maybe accompanied by mentioning the basic categories like Authentication & Security, Asset. Negotiation, Transfer, Policy Engine, Protocol?

Author: How about this: we outline what "plugpoints" we have, i.e. where devs can hook in and extend/customize their runtimes? Those would include:
- Authentication/Authorization of the Management API
- policies and evaluation functions
- persistence implementations
- provisioners
- transaction management (?)

Replacing the Protocol (DSP) is a _really_ advanced topic, which I would not put in the developer handbook.

--> how to write your own first extension, explains plug-points points, metamodel annotations, SPIs, etc

> ### Dependency Injection à la EDC

Reviewer 2: ```suggestion
### EDC dependency injection
```
